src/panels/lovelace/editor/view-editor/hui-view-visibility-editor.ts

```diff
@@ -0,0 +1,156 @@
+import {
+  html,
+  LitElement,
+  TemplateResult,
+  customElement,
+  property,
+  PropertyValues,
+  CSSResult,
+  css,
+} from "lit-element";
+import "@polymer/paper-input/paper-input";
+
+import { HomeAssistant } from "../../../../types";
+import { fireEvent } from "../../../../common/dom/fire_event";
+import { configElementStyle } from "../config-elements/config-elements-style";
+import { LovelaceViewConfig, ShowViewConfig } from "../../../../data/lovelace";
+
+import { fetchUsers, User } from "../../../../data/user";
+import memoizeOne from "memoize-one";
+import { compare } from "../../../../common/string/compare";
+import { HaSwitch } from "../../../../components/ha-switch";
+
+declare global {
+  interface HASSDomEvents {
+    "view-visibility-changed": {
+      visible: ShowViewConfig[];
+    };
+  }
+}
+
+@customElement("hui-view-visibility-editor")
+export class HuiViewVisibilityEditor extends LitElement {
+  set config(config: LovelaceViewConfig) {
+    this._config = config;
+    this._visible = this._config.visible;
+  }
+
+  @property() public hass!: HomeAssistant;
+  @property() public _config!: LovelaceViewConfig;
+  @property() private _users!: User[];
+  @property() private _visible?: boolean | ShowViewConfig[];
+
+  private _sortedUsers = memoizeOne((users: User[]) => {
+    return users
+      .filter((user) => !user.system_generated)
+      .sort((a, b) => compare(a.name, b.name));
+  });
+
+  protected firstUpdated(changedProps: PropertyValues) {
+    super.firstUpdated(changedProps);
+
+    fetchUsers(this.hass).then((users) => {
+      this._users = users;
+      // how can I resize and reposition dialog? Should I use `iron-resize` event?
+      // fireEvent(this as HTMLElement, "iron-resize");
```

Comment:
If I uncomment this then the dialog flashes, if I leave this then the dialog isn't positioned correctly

Reply:
You should fire `iron-resize`

Reply:
If I fire it then the form blinks. Please see https://github.com/home-assistant/frontend/pull/4978#issuecomment-592010929.
I know this is because of async functionality, but how can I deffer rendering until users get loaded? Is there a function I could use for that?

Reply:
No, the dialog is already there, and you don't want to wait with reacting on the click till the users are loaded.

Reply:
ok, but what about params? Should it be `fireEvent(this as HTMLElement, "iron-resize");`?

Reply:
yes, but the `as HTMLElement` is not needed.

Reply:
fixed